silx/io/__init__.py

```diff
@@ -36,6 +36,10 @@
 from .utils import is_file
 from .utils import is_group
 from .utils import is_softlink
+from .utils import supported_extensions
 
 # avoid to import open with "import *"
-__all__ = ["save1D", "is_dataset", "is_file", "is_group", "is_softlink"]
+__all = locals().keys()
+__all = filter(lambda x: not x.startswith("_"), __all)
+__all = filter(lambda x: x != "open", __all)
+__all__ = list(__all)
```

Comment:
I personally preferred the previous `__all__` definition even if it needs to be maintained by hand.

Reply:
This do what it is supposed to do. Filtering out open.
The previous way is not maintainable, and will grow for every.